libraries/AP_Compass/AP_Compass_MAG3110.cpp

```diff
@@ -0,0 +1,291 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <utility>
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include <stdio.h>
+
+#include "AP_Compass_MAG3110.h"
+
+extern const AP_HAL::HAL &hal;
+
+
+/*
+    на первый взгляд, магнитометр MAG3110 состоит из одних лишь недостатков:
```

Comment:
Please put comments in English

Reply:
My english is very poor, it will be better that someone with native English will write description of this driver. I can add english version without removing russian, will it be OK?